### ISSUE TITLE
Challenge #34 - Create the shortest palindrome from a string

### ISSUE THREAD

**Issue by DominiqueCaron** [DominiqueCaron/daily-coding-problem#7]
# The Challenge

Given a string, find the palindrome that can be made by inserting the fewest number of characters as possible anywhere in the word. If there is more than one palindrome of minimum length that can be made, return the lexicographically earliest one (the first one alphabetically).

For example, given the string "race", you should return "ecarace", since we can add three letters to it (which is the smallest amount to make a palindrome). There are seven other palindromes that can be made from "race" by adding three letters, but "ecarace" comes first alphabetically.

As another example, given the string "google", you should return "elgoogle".

**Comment by juanmanual** [comment]
This seems equivalent to finding the longest palindrome on either end, then adding the remaining letters.

**Comment by DominiqueCaron** [comment]
Since we can insert characters anywhere in the word, I do not think this is true in all cases. 

It works for the two examples given in the challenge, but it is not always the case. For example, for "abca", the function should return "abcba", I think.

**Comment by juanmanual** [comment]
True. I didn't see that in the examples so I must have glossed over when reading the problem description.
